dtfs-central-api/src/v1/controllers/tfm/deal/tfm-update-deal.controller.js

```diff
@@ -42,6 +47,27 @@ const updateDeal = async (dealId, dealChanges, existingDeal) => {
     }
   }
 
+  /**
+   * Ensure tfm.activities is not wiped and avoid recursive object creation
+   * by checking .type property for the activities object
+   * */
+  if (existingDeal.tfm && existingDeal.tfm.activities && tfmUpdate.tfm.activities.type) {
+    dealUpdate = {
+      tfm: {
+        ...existingDeal.tfm,
+        ...tfmUpdate.tfm,
+        activities: existingDeal.tfm.activities,
+      },
+    };
+
+    if (tfmUpdate.tfm.activities) {
+      dealUpdate.tfm.activities = [
+        ...existingDeal.tfm.activities,
+        tfmUpdate.tfm.activities,
+      ];
+    }
+  }
+
```

Comment:
Need to add a test for this, the endpoint has some tests in tfm-deal-put-tfm.api-test.js